components/keyrings/common/config/config_reader.h

```diff
@@ -43,7 +43,19 @@ class Config_reader {
     @param [in] config_file_path Full path to configuration file
   */
   explicit Config_reader(const std::string config_file_path);
+#ifdef XTRABACKUP
+  /**
+    Constructor
+
+    Reads JSON from parameter and stores it in memory.
 
+    @param [in] config_file_path Full path to configuration file - not used
```

Comment:
Not used?  incase of offline (--prepare), we pass the file here? :thinking:

Reply:
We pass the config_data already populated,  from keyring config.cc - https://github.com/altmannmarcelo/percona-xtrabackup/blob/PXB-2657-8.0/components/keyrings/keyring_file/config/config.cc#L90, here we are just overloading the constructor to not try to load the manigest config file as they won't be on their set location  - https://dev.mysql.com/doc/refman/8.0/en/keyring-component-installation.html .

Problem here is that config_data and config_file_path are both of the same type, I might be able to adjust the Constructor to accept `rapidjson::Document` so we can overload it with a single parameter.

Reply:
Done